addons/mail/tests/test_mail_canned_response.py

```diff
@@ -0,0 +1,107 @@
+        answers or the one linked to the user group """
+        for test_user, expected_responses in [
+            (
+                self.user_employee,
+                self.response_admin_group_user + self.response_user_group_user + self.response_user
+            ),
+            (
+                self.user_admin,
+                # response_user is limited to creator, even admin does not receive it
+                self.response_admin_group_user + self.response_admin_group_system + self.response_user_group_user
+            ),
+            (
+                self.response_admin,
+                # even with full access, see only own + groups
+                self.response_admin_group_user + self.response_user_group_user
+            ),
+        ]:
+            with self.subTest(user_login=test_user.login):
+                self.authenticate(test_user.login, test_user.login)
+                request = self.make_jsonrpc_request("/mail/data", {"canned_responses": True})
+                expected_content = [
+                    {
+                        'id': cr.id,
+                        'source': cr.source,
+                        'substitution': cr.substitution
+                    } for cr in expected_responses
+                ]
+                self.assertListEqual(request["CannedResponse"], expected_content)
+
+    def test_mail_canned_response_permissions(self):
+        """ Test ACLs and security on 'mail.canned.response' model, notably that
+        group_ids restrict edition """
+        for test_user, write_ok in [
+            (self.user_employee, self.response_user + self.response_user_group_user),
+            (self.user_admin, self.response_admin_group_system + self.response_admin_group_user),
+            (self.response_admin, self.response_all),
+        ]:
+            for canned_response in self.response_all:
+                with self.subTest(user_login=test_user.login, response_name=canned_response.source):
+                    canned_response_as_user = canned_response.with_user(test_user)
+                    if canned_response in write_ok:
+                        canned_response_as_user.write({"substitution": f"{canned_response.source} UPDATED"})
+                    else:
+                        with self.assertRaises(AccessError):
+                            canned_response_as_user.write({"substitution": f"{canned_response.source} UPDATED"})
+
+        employee_unlink_ok = self.response_user + self.response_user_group_user
+        for canned_response in self.response_all:
+            with self.subTest(response_name=canned_response.source):
+                canned_response_as_user = canned_response.with_user(self.user_employee)
+                if canned_response in employee_unlink_ok:
+                    canned_response_as_user.unlink()
+                else:
+                    with self.assertRaises(AccessError):
+                        canned_response_as_user.unlink()
```

Comment:
Is the `subTest` resetting the transaction?
Otherwise previously deleted one would immediately resolve before checking ACL on following calls.